crates/typst/src/lib.rs

```diff
@@ -231,6 +231,48 @@ pub trait World {
     }
 }
 
+macro_rules! delegate_for_ptr {
+    ($W:ident for $ptr:ty) => {
+        impl<$W: World> World for $ptr {
```

Comment:
It seems that `$W` is unused

Reply:
- `$W:ident` defines the metavar
- `<$W: World>` uses the metavar and defines the generic parameter
- `$ptr` uses the generic parameter

Reply:
Oh, got it